### ISSUE TITLE
Plugin is amazing, but getting buggy

### ISSUE THREAD

**Issue by Aaronik** [jackMort/ChatGPT.nvim#124]
Hello,

Your plugin is awesome - I mean, it's my favorite AI plugin I've found so far for neovim, so thank you!

I'm experiencing some bugs, though -- for example, scrolling pops up a fat error:

![image](https://user-images.githubusercontent.com/1324601/227681730-b7c377a7-8bfc-426c-9980-f9304efa540b.png)

And I have to do a `:bd` to quit out of the `ChatGPTEditWithInstructions` window (it doesn't respect the key commands the others do)

Also, with `ChatGPTEditWithInstructions`, it only occasionally wants to accept the text I've copied.

The sizing of the input box has very strange behavior, sometimes it expands to fit the text inside of it, sometimes it doesn't, and it's not clear how to scroll when it doesn't.

All in all I really love this plugin and want to see it succeed - but right now it's starting to get a little buggy. I've seen amazing projects go downhill at this phase of things, when they try to do a lot of cool stuff, but they don't focus on the stability. I was wondering -- do you plan on putting in place any **tests**? Other amazing authors like folke [have tests](https://github.com/folke/lazy.nvim/tree/main/tests) for their plugins. I wonder if having a test suite might both help to make your amazing plugin have a little bit more rounded edges, and simultaneously help take some stress off your shoulders about the plugin's stability.

Just 2 cents, thanks again for your plugin!

**Comment by DGmip** [comment]
Yeah I agree, it's getting a little frustrating to have to quit my entire buffer all the time, it's making it a bit of a risk to use at times and I've started making sure I save before I open it

**Comment by cmpadden** [comment]
Scrolling has been resolved in PR https://github.com/jackMort/ChatGPT.nvim/pull/130. I hope to look into the inconsistent exit behavior as I get some bandwidth.

I think having tests is a great idea,  @Aaronik. If you wanted to take an initial stab at it, I'd be happy to collaborate.
